core/io/resource_loader.cpp

```diff
@@ -316,14 +316,34 @@ Ref<Resource> ResourceLoader::_load(const String &p_path, const String &p_origin
 		return res;
 	}
 
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
```

Comment:
Restricting this to editor sessions (and not just editor builds) because at runtime many types are not registered with the importer and therefore the check below would be less reliable. That way, we have two behaviors only: editor and runtime, and not an in-between third one, that would make things unnecessarily complex.